src/eduid/userdb/tests/test_userdb.py

```diff
@@ -93,49 +94,63 @@ def test_get_user_by_eppn_not_found(self):
         assert self.amdb.get_user_by_eppn("abc123") is None
 
 
-class UserMissingMeta(MongoTestCaseRaw):
+class UserMissingMeta(MongoTestCase):
+    user: User
+
     def setUp(self, *args, **kwargs):
-        self.user = self._raw_user()
-        self.time_now = utc_now()
-        super().setUp(raw_users=[self.user])
+        self.user = UserFixtures().mocked_user_standard
+        super().setUp(*args, am_users=[self.user], **kwargs)
 
-    def _raw_user(self) -> Dict[str, Any]:
-        user = mocked_user_standard.to_dict()
-        del user["meta"]
-        return user
+        self._remove_meta_from_user_in_db(self.user)
+
+    def _remove_meta_from_user_in_db(self, user: User) -> None:
+        """
+        These tests are meant to test new code with old users in the database.
+
+        Remove the user.meta section from the user in the database.
+        """
+        user_doc = self.amdb._get_document_by_attr("_id", user.user_id)
+        assert user_doc is not None
+        user_doc.pop("meta")
+        self.amdb._coll.replace_one({"_id": user.user_id}, user_doc)
 
     def test_update_user_new(self):
-        db_user = self.amdb.get_user_by_id(self.user["_id"])
+        db_user = self.amdb.get_user_by_id(self.user.user_id)
+        assert db_user is not None
+        logger.debug(f"Loaded user.meta from database:\n{format_dict_for_debug(db_user.meta.dict())}")
+        assert db_user is not None
         db_user.given_name = "test"
-        self.amdb.save(user=db_user, check_sync=True)
+        self.amdb.save(user=db_user)
 
     def test_update_user_old(self):
-        db_user = self.amdb.get_user_by_id(self.user["_id"])
+        db_user = self.amdb.get_user_by_id(self.user.user_id)
         db_user.given_name = "test"
-        self.amdb.old_save(user=db_user, check_sync=True)
+        self.amdb.save(user=db_user)
 
 
-class UpdateUser(MongoTestCaseRaw):
+class UpdateUser(MongoTestCase):
     def setUp(self, *args, **kwargs):
-        self.user = mocked_user_standard
-        super().setUp(am_users=[self.user, mocked_user_standard_2], **kwargs)
+        _users = UserFixtures()
+        self.user = _users.mocked_user_standard
+        super().setUp(am_users=[self.user, _users.mocked_user_standard_2], **kwargs)
 
     def test_stale_user__meta_version(self):
```

Comment:
dubbelt understräck har smugit sig in, antagligen mitt fel från tidigare revition :)